runtime/src/main/scala/akka/grpc/scaladsl/headers/headers.scala

```diff
@@ -81,19 +87,86 @@ object `Status` extends ModeledCustomHeaderCompanion[`Status`] {
     headers.collectFirst { case h if h.is(name) => Integer.parseInt(h.value()) }
 }
 
-// TODO percent-encoding of message?
-final class `Status-Message`(override val value: String) extends ModeledCustomHeader[`Status-Message`] {
+// grpc-message must be percent encoded: https://github.com/grpc/grpc/issues/4672
+final class `Status-Message`(val unencodedValue: String) extends ModeledCustomHeader[`Status-Message`] {
+  import `Status-Message`._
+
   override def renderInRequests = false
   override def renderInResponses = true
   override val companion = `Status-Message`
+  override def value() = toAsciiString(unencodedValue)
+
+  // Copied with slight adaptations from https://github.com/grpc/grpc-java/blob/79e75bace40cea7e4be72e7dcd1f41c3ad6ee857/api/src/main/java/io/grpc/Status.java#L582
+  private def toAsciiString(value: String): String = {
+    val valueBytes = value.getBytes(StandardCharsets.UTF_8)
+    val firstIndexToEscape = valueBytes.indexWhere(isEscapingChar(_))
+    if (firstIndexToEscape == -1)
+      new String(valueBytes)
+    else
+      toAsciiStringSlow(valueBytes, firstIndexToEscape)
+  }
+
+  private def isEscapingChar(b: Byte): Boolean = b < ' ' || b >= '~' || b == '%'
+
+  private def toAsciiStringSlow(valueBytes: Array[Byte], riArg: Int): String = {
```

Comment:
move `toAsciiString`, `isEscapingChar` and `toAsciiStringSlow` to companion object? They seem to be all pure.